improver_tests/utilities/temporal/test_TimezoneExtraction.py

```diff
@@ -0,0 +1,263 @@
+        plugin.check_input_cube_dims(cube)
+    else:
+        with pytest.raises(
+            ValueError, match=r"Expected coords on input_cube: time, y, x "
+        ):
+            plugin.check_input_cube_dims(cube)
+
+
+@pytest.mark.parametrize(
+    "local_time, expect_success",
+    ((datetime(2017, 11, 10, 5, 0), True), (datetime(2017, 11, 10, 6, 0), False)),
+)
+def test_check_input_cube_time(local_time, expect_success):
+    """Checks that check_input_cube_time can differentiate between arguments that match
+    expected times and arguments that don't."""
+    data_shape = [3, 3]
+    cube = make_input_cube(data_shape)
+    timezone_cube = make_timezone_cube()
+    plugin = TimezoneExtraction()
+    if expect_success:
+        plugin.check_input_cube_time(cube, timezone_cube, local_time)
+    else:
+        with pytest.raises(
+            ValueError, match=r"Time coord on input cube does not match required times."
+        ):
+            plugin.check_input_cube_time(cube, timezone_cube, local_time)
+
+
+def test_check_timezones_are_unique_pass():
+    """Checks that check_timezones_are_unique allows our test cube"""
+    timezone_cube = make_timezone_cube()
+    plugin = TimezoneExtraction()
+    plugin.check_timezones_are_unique(timezone_cube)
+
+
+@pytest.mark.parametrize("offset", (1, -1))
+def test_check_timezones_are_unique_fail(offset):
+    """Checks that check_timezones_are_unique fails if we break our test cube"""
+    timezone_cube = make_timezone_cube()
+    timezone_cube.data[0, 0, 0] += offset
+    plugin = TimezoneExtraction()
+    with pytest.raises(
+        ValueError,
+        match=r"Timezone cube does not map exactly one time zone to each spatial point",
+    ):
+        plugin.check_timezones_are_unique(timezone_cube)
+
+
+@pytest.mark.parametrize("input_as_cube", (True, False))
+@pytest.mark.parametrize("input_has_time_bounds", (True, False))
+def test_process(input_as_cube, input_has_time_bounds):
+    """Checks that the plugin process method returns the a cube with expected data and
+    time coord for our test data"""
+    data_shape = [3, 3]
+    cube = make_input_cube(data_shape, time_bounds=input_has_time_bounds)
+    data = np.array(
+        [
+            np.full((3, 3), fill_value=-1, dtype=np.float32),
+            np.zeros((3, 3), dtype=np.float32),
+            np.ones((3, 3), dtype=np.float32),
+        ]
+    )
+    cube.data = data
+    local_time = datetime(2017, 11, 10, 5, 0)
+    timezone_cube = make_timezone_cube()
+    expected_data = [[-1, -1, -1], [0, 0, 0], [1, 1, 1]]
+    row1 = [cube.coord("time").units.date2num(datetime(2017, 11, 10, 4, 0))] * 3
+    row2 = [cube.coord("time").units.date2num(datetime(2017, 11, 10, 5, 0))] * 3
+    row3 = [cube.coord("time").units.date2num(datetime(2017, 11, 10, 6, 0))] * 3
+    expected_times = [row1, row2, row3]
+    expected_bounds = np.array(expected_times).reshape((3, 3, 1)) + [[[-3600, 0]]]
+    plugin = TimezoneExtraction()
```

Comment:
Hmm. Possibly. Not sure where I'd put it though as these tests are pytests, not unittests. I'd have to create a function to do this and it would still need one line to call it. Instead, I've changed all the tests that only use `plugin` once, and in-lined the plugin initiation.

Reply:
Ah of course, I was thinking of unit tests, yes.